### PR TITLE
feat: use tryBlockAndAggregate to fetch balances

### PR DESCRIPTION
**Pull request by mathiasmoeller**
My idea was to isolate broken tokens by chunking down to calls of 1 to make sure we get all the other tokens.
Not sure if the log is the best idea but I didn't see another good way how to inform about the broken token.

### CODE REVIEW

File: src/balances/index.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ const getTokenBalance = async (
   walletAddress: string,
   token: Token
 ): Promise<TokenAmount | null> => {
+  debugger
```
Comment (H3xept) [inline src/balances/index.ts]: Do we really want this?

File: src/utils/multicall.ts [hunk 2]
```diff
@@ -90,10 +81,56 @@ export const fetchDataUsingMulticall = async (
...
+          )
+          return []
+        } else {
+          // try with chunk size of one to identify exactly the failed call
+          const smallestChunks = await chunkDataAndExecuteMulticall(
+            abiInterface,
+            multicallContract,
+            calls,
+            chainId,
+            1
```
Comment (H3xept) [inline src/utils/multicall.ts:98]: Correct me if I'm wrong but if a large chunk fails, this code will re-chunk that chunk with chunk size == 1 and re execute? 

If so, would it maybe be better to recurse by calling chunkSize / 2 ? This guarantees to reach the failing token in O(log(n)) calls and we won't waste bandwith / API call allowance / time in doing a ton of size 1 multicall calls.

**Comment by H3xept** [comment]
I left my comments before seeing max's. He's right but I think tryAggregate does not allow us to find out which token is problematic?

**Comment by Addminus** [comment]
I think we need both multicall`v2` and a solution for `v1`. I don't think `v2` is deployed to all chains we need/support

File: src/balances/index.ts (hunk 1 above)
Comment (mathiasmoeller) [reply src/balances/index.ts]: This was removed loooong before your review :P

**Comment by H3xept** [comment]
> I think it does, it returns a list of responses where the failing on is empty, so we can filter for that one and find the matching address.

Ok yeah the multicall contract returns a list of results which can also be not successful https://github.com/makerdao/multicall/blob/master/src/Multicall2.sol#L64

**Review by maxklenk — APPROVED** [review]

**Pull request merged**